connect/templates/connect-deployment.yaml

```diff
@@ -40,6 +40,14 @@ spec:
         - name: {{ .Values.connect.api.name }}
           image: {{ .Values.connect.api.imageRepository }}:{{ .Values.connect.version | default .Chart.AppVersion }}
           imagePullPolicy: {{ .Values.connect.imagePullPolicy }}
+          securityContext:
+            runAsUser: 999
+            runAsGroup: 999
+            allowPrivilegeEscalation: false
+            capabilities:
+              add:
+              - "NET_ADMIN"
+              - "NET_BROADCAST"
```

Comment:
Is there a specific reason for adding this? If I understand the k8s security model correctly, this _adds_ `CAP_NET_ADMIN` and `CAP_NET_BROADCAST` to the default set of capabilities. Looking at the [Linux kernel docs](https://man7.org/linux/man-pages/man7/capabilities.7.html), I am not sure if we need those caps.

Reply:
The [security context docs](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) give the example of adding `CAP_NET_ADMIN` and `CAP_SYS_TIME` capabilities as:

```
    securityContext:
      capabilities:
        add: ["NET_ADMIN", "SYS_TIME"]
```

So I followed the docs on this

Reply:
I'm sorry, I wasn't really clear. I mean: is there a reason we need `NET_ADMIN` and `NET_BROADCAST`? Because it seems the containers have been running fine without and looking at the definitions of those capabilities, it does not sounds like they allows something we need:

 `CAP_NET_ADMIN`
Perform various network-related operations:
* interface configuration;
* administration of IP firewall, masquerading, and accounting;
* modify routing tables;
* bind to any address for transparent proxying;
* set type-of-service (TOS);
* clear driver statistics;
* set promiscuous mode;
* enabling multicasting;
* use setsockopt(2) to set the following socket options:  SO_DEBUG, SO_MARK, SO_PRIORITY (for a priority outside the range 0 to 6), SO_RCVBUFFORCE, and SO_SNDBUFFORCE.

`CAP_NET_BROADCAST`
(**Unused**)  Make socket broadcasts, and listen to multicasts.

Reply:
I included NET_ADMIN specifically because of `enabling multicasting;` since the containers need to be able to broadcast multicast, not just listen

Reply:
Just did a quick test and `NET_ADMIN` is not required. Also I included `drop: ["all"]` to further reduce the listed capabilities in `/proc/1/status`